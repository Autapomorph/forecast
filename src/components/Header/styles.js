import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledHeader = styled.header`
  grid-area: header;
  display: flex;
  justify-content: center;
  min-width: 320px;
  padding-top: 30px;
`;
