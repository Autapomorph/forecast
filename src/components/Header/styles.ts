import styled from 'styled-components/macro';

export const StyledHeader = styled.div`
  min-width: 320px;
  padding-top: 30px;
`;
