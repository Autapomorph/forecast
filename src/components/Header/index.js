import React from 'react';

import SearchBar from '../SearchBar';
import UnitsFormats from '../UnitsFormats';
import Languages from '../Languages';

import { StyledHeader } from './styles';

const Header = () => (
  <StyledHeader>
    <SearchBar />
    <Languages />
    <UnitsFormats />
  </StyledHeader>
);

export default Header;
