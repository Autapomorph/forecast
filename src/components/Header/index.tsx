import React from 'react';

import SearchBar from 'components/SearchBar';

import * as S from './styles';

const Header = (): React.ReactElement => (
  <S.Header>
    <SearchBar />
  </S.Header>
);

export default Header;
