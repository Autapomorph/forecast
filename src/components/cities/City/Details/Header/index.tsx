import React from 'react';
import { useTranslation } from 'react-i18next';

import { City, Weather } from 'models';
import generateName from 'utils/city/generateName';
import { toDayMonthHourMinutes } from 'utils/weather/time/converters';
import Title from 'components/common/Title';
import CountryFlag from 'components/common/icons/CountryFlag';
import RetryButton from 'components/common/buttons/Retry';
import FeaturedButton from 'components/common/buttons/Featured';
import MarginRightDelimeter from 'components/common/delimeters/MarginRight';

import * as S from './styles';

type Props = {
  city: City & Weather;
  isFeatured: boolean;
  refetchWeather: () => void;
  addToFeatured: (city: City) => void;
  removeFromFeatured: (cityId: City['id']) => void;
};

const CityHeader = ({
  city,
  isFeatured,
  refetchWeather,
  addToFeatured,
  removeFromFeatured,
}: Props): React.ReactElement => {
  const { i18n } = useTranslation();

  return (
    <S.CityHeader>
      <Title>
        <span>
          <span>{`${generateName(city)}, `}</span>
          <span style={{ whiteSpace: 'nowrap' }}>
            {city.countryName}
            <MarginRightDelimeter />
            <CountryFlag country={city.countryCode.toLowerCase()} />
          </span>
        </span>
        <Title.Subtitle>
          {toDayMonthHourMinutes(city.current.timestamp, i18n.language)}
        </Title.Subtitle>
      </Title>

      <S.TitleButtonsWrapper>
        <RetryButton onClick={refetchWeather} />
        <FeaturedButton
          isFeatured={isFeatured}
          onRemove={() => removeFromFeatured(city.id)}
          onAdd={() =>
            addToFeatured({
              id: city.id,
              name: city.name,
              region: city.region,
              countryName: city.countryName,
              countryCode: city.countryCode,
              coords: city.coords,
            })
          }
        />
      </S.TitleButtonsWrapper>
    </S.CityHeader>
  );
};

export default CityHeader;
