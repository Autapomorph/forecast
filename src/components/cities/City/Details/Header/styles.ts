import styled from 'styled-components/macro';

import { Title } from 'components/common/Title/styles';

export const CityHeader = styled.div`
  grid-area: city-title;
  display: flex;
  align-items: baseline;

  ${Title} {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
  }
`;

export const TitleButtonsWrapper = styled.div`
  flex-shrink: 0;
`;
