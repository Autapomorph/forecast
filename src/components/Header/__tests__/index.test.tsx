import React from 'react';
import { shallow } from 'enzyme';

import Header from '..';

describe('Header tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
