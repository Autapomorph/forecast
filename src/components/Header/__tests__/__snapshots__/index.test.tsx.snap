// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders properly 1`] = `
<styles__Header>
  <Connect(SearchBar) />
</styles__Header>
`;
